### DISCUSSION TITLE
Create ebsd file from simulated dataset

### DISCUSSION THREAD

**Discussion by mohammadsattari1990** [mtex-toolbox/mtex#1986] [Ask Anything]
I have simulated grain formation during the solidification process of cubic metals, using the cellular automata method. The output of simulations is a data set of points, and corresponding values, including:
- coordinates
- phase
- grain index
- grain orientations with 3 Euler angles
- ...
Is it possible to create an "ebsd" object, then file (e.g. *.ctf) from such data using MTEX? If not, it would be valuable to add such a tool.

**Comment by kilir** [comment]
Hi,
if you plan to work in mtex. you can directly create an EBSD or grain object without the need to write it to some format such as .ctf. More importantly, are the coordinates points of such as EBSD or are those vertices of grains or somethign completely different? 
Maybe you can provide an example of such data.
Cheers,
Rüdiger

**Reply by mohammadsattari1990** [reply]
[Simulation Outputs.zip](https://github.com/mtex-toolbox/mtex/files/13078211/Simulation.Outputs.zip)

Hi Rüdiger,
Attached is a MAT file having a structure, CELL, that contains
- center_coords: grid center coordinates,
- grain_idx: grain index of each grid
- ori: orientation of each grid by 3 Euler angles
- ori_convection: rotation convection, here ZXZ
- cryssym: crystallographic symmetry family
- phases: string of name of present phases
- phase: phase of each grid
- grid_size: grid size
- section_axis: name of 2 axis of the current cross-section, here Y,Z

A *.m file is also attached to calculate the vertices of grids, each row of "vertice_coords" represents [x1,y1,x2,y2,x3,y3,x4,y4] in a counterclockwise representation.
